sgkit/distance/api.py

```diff
@@ -94,12 +98,21 @@ def pairwise_distance(
            [ 2.62956526e-01,  0.00000000e+00,  2.14285714e-01],
            [ 2.82353505e-03,  2.14285714e-01,  0.00000000e+00]])
     """
+    valid_targets = TargetTypes.__args__  # type: ignore[attr-defined]
```

Comment:
I like this but it's unprecedented within the rest of the codebase AFAIK.  I'm weakly in favor of not doing checks like this and relying on mypy instead since that's what we do in say https://github.com/pystatgen/sgkit/blob/master/sgkit/stats/aggregation.py (see `Dimension`).  I think it is optimistic to expect that any of our users care about running mypy on their code, but we should pick a consistent idiom regardless (and add an issue to use this kind of check more often if we go this route).

Do you have any thoughts on this @ravwojdyla?